lib/octokit/client/deployments.rb

```diff
@@ -1,69 +1,87 @@
+      # @param options [String] :task The name of the task for the deployment (e.g., `deploy` or `deploy:migrations`).
+      # @param options [String] :environment The name of the environment that was deployed to (e.g., `staging` or `production`).
       # @return [Array<Sawyer::Resource>] A list of deployments
       # @see https://developer.github.com/v3/repos/deployments/#list-deployments
       def deployments(repo, options = {})
         get("#{Repository.path repo}/deployments", options)
       end
       alias :list_deployments :deployments
 
-      # Create a deployment for a ref
+      # Create a deployment
       #
       # @param repo [Integer, String, Repository, Hash] A GitHub repository
-      # @param ref [String] The ref to deploy
-      # @option options [String] :task Used by the deployment system to allow different execution paths. Defaults to "deploy".
-      # @option options [String] :payload Meta info about the deployment
-      # @option options [Boolean] :auto_merge Optional parameter to merge the default branch into the requested deployment branch if necessary. Default: true
-      # @option options [Array<String>] :required_contexts Optional array of status contexts verified against commit status checks.
-      # @option options [String] :environment Optional name for the target deployment environment (e.g., production, staging, qa). Default: "production"
-      # @option options [String] :description Optional short description.
-      # @return [Sawyer::Resource] A deployment
+      # @param ref [String] The ref to deploy. This can be a branch, tag, or SHA.
+      # @param options [String] :task Specifies a task to execute (e.g., `deploy` or `deploy:migrations`).
+      # @param options [Boolean] :auto_merge Attempts to automatically merge the default branch into the requested ref, if it is behind the default branch.
+      # @param options [Array<string>] :required_contexts The status contexts to verify against commit status checks. If this parameter is omitted, then all unique contexts will be verified before a deployment is created. To bypass checking entirely pass an empty array. Defaults to all unique contexts.
+      # @param options [String] :payload JSON payload with extra information about the deployment.
+      # @param options [String] :environment Name for the target deployment environment (e.g., `production`, `staging`, `qa`).
+      # @param options [String] :description Short description of the deployment.
+      # @param options [Boolean] :transient_environment Specifies if the given environment is specific to the deployment and will no longer exist at some point in the future. **This parameter requires a custom media type to be specified.**
+      # @param options [Boolean] :production_environment Specifies if the given environment is one that end-users directly interact with. **This parameter requires a custom media type to be specified.**
```

Comment:
This looks like a big block of changes but mostly it's two things:

- the descriptions changed
- there are some additional optional arguments that weren't listed before.

The arguments that are under preview had some kind of weird (in this context) descriptions that said "see note below". I hard-coded a fix to remove that.